cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -589,7 +589,8 @@ public void createApplicationVersion(ApplicationId id,
                                     appMeta.getChange().getCreationTimeMillis() - 1000));
       }
       fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "false"));
-      getApplicationSpecificationTable().upsert(fields);
+      // Correctly update the previous latest version
+      getApplicationSpecificationTable().update(fields);
```

Comment:
`insert` issue does not exist with the DB isolation level we have. What do you mean by `backwards compatible`? Since here we made sure the row exists, the correct table method should be `update`, right?

Reply:
I meant this doesn't affect FF off case

Reply:
For FF off case, it still should be `update`, no? Could you elaborate?

Reply:
you are right, my bad

Reply:
no worries at all

Reply:
I reverted it back... because noSQL update is not updating the null index field...